### DISCUSSION TITLE
Rules - how do I set up?

### DISCUSSION THREAD

**Discussion by CyBaart** [arendst/Tasmota#19312] [Q&A]
I must be being particularly thick, but no matter how much googling I do, I cant get started with rules. I have both an ESP32C3 and an ESP32 devkit with the latest FW installed. Is there no option on the menu to write rules of is it the 'Berry scripting' OR can I use Berry Scripting. Basically, I have an 18DS20B sensor for water temperature, a DHT11 for ambient air temperature and humidity and three solid state relays. They all work on the console, I get correct values and can toggle the relays on and off. This is for my hot tub. The relays are for circ pump, main control panel (when energised the main pump comes on low speed. I can turn the pump on high speed, turn the blowers on, and turn the LED lighting on but I have removed the heater control as I want to make some rules eg during the day when the air temperature is 30+°C I want the water cool. On a night when the air temperature drops (I live in the mountains and the nights are cool even in the height of summer - btw, I live in Cyprus) I want the water warm.

So, please someone get me started on rules - not how to write them, thats easy - where I write them, save them, initiate them.

Thanks, Steve

**Comment by pkkrusty** [comment]
In the console. (Not Berry console). Write the rules in one of three rulesets, then activate the rule set. 

Eg `rule1 on ds18b20#temperature>30 do power1 1 endon on ds18b20#temperature<=29 do power1 0 endon` then `rule1 1`

You won't initiate them. They initiate via triggers. You can however, turn them on and off.

**Reply by CyBaart** [reply]
So, I just type the rule into the command field of the console. Thank you, I don't know why the tutorials don't make that plain.

Thank you again, Steve.
